### ISSUE TITLE
protocol switching

### ISSUE THREAD

**Issue by anbu-squad** [UnamSanctam/SilentCryptoMiner#2281]
How does the API switch between protocols?
http and https

**Comment by UnamSanctam** [comment]
The library cpp-httplib does it automatically depending on if the URL contains `http://` or `https://`

**Comment by anbu-squad** [comment]
```
#define CPPHTTPLIB_OPENSSL_SUPPORT
#include "httplib.h"

```
GPT chat tells me that it is also needed OpenSSL

this is right?


and the second question, to check links for accessibility using the library, do they need to be parsed and the domain or IP itself checked for accessibility?

**Comment by UnamSanctam** [comment]
Yes you can see the examples here: https://github.com/yhirose/cpp-httplib. When you do a post or get request then they will return a status code just like a normal browser, for example any status code between 200 and 400 (including 200) is usually a successful request.
